datasets/ae/local_terrorists/crawler.py

```diff
@@ -27,15 +35,22 @@ def parse_row(
         if header in ["index", "issuer"]:
             continue
         if type_ == "date":
-            value = h.parse_date(value, ["%d/%m/%Y"])
+            value = h.extract_date(context.dataset, value)
         if header == "category":
             schema = context.lookup_value("categories", value)
             if schema is None:
                 context.log.error("Unknown category", category=value)
             elif not entity.schema.is_a("Vessel"):
                 entity.add_schema(schema)
             continue
-        if header in ("program", "listingDate", "endDate", "provisions"):
+        if header in ("program", "provisions"):
+            sanction.add(header, value, lang=lang)
+            continue
+        if header in ("listingDate", "endDate"):
+            for date in DATES:
+                match = date.match(value[0])
+                if match is not None:
+                    value = match.group(1)
             sanction.add(header, value, lang=lang)
```

Comment:
how about not parsing it above at `if type_ == date`, 

then here at `if header in ("listingDate...`, instead of trying to match, you just replace "مدرج بموجب قرار مجلس الوزراء رقم" and the other one with "" and calling `apply_date(sanction, ...)`, 

then you can handle `\(\d+\) لسنة (\d{4})` using `Dataset.dates.formats`

it's a bit simpler and avoids an unnecessary h.extract_date